Registration/RegisterWithAzure.psm1

```diff
@@ -1649,56 +1640,47 @@ function Get-AzToken
+            if (-not ([Microsoft.Azure.Commands.Common.Authentication.AzureSession]::Instance.TryGetComponent(
+                [Microsoft.Azure.Commands.Common.Authentication.Authentication.Clients.AuthenticationClientFactory]::AuthenticationClientFactoryKey,
+                [ref]$authenticationClientFactory)))
+            {
+                $m = 'Please ensure you have authenticated with Az Accounts module!'
+                $m += ' Unable to resolve authentication client factory from Az Accounts module runtime'
+                $m += ' ([Microsoft.Azure.Commands.Common.Authentication.Authentication.Clients.AuthenticationClientFactory])'
+                Write-Error $m
+                return
+            }
+
+            $client = $authenticationClientFactory.CreatePublicClient(
+                ($clientId='1950a258-227b-4e31-a9cf-717495945fc2'),
+                ($TenantId),
+                ($authority="$($Context.Environment.ActiveDirectoryAuthority.TrimEnd('/'))/$TenantId"),
+                ($redirectUri='urn:ietf:wg:oauth:2.0:oob'),
+                ($useAdfs=$Context.Environment.ActiveDirectoryAuthority -like '*/adfs*'))
+
+            $authenticationClientFactory.RegisterCache($client)
+            $accounts = $client.GetAccountsAsync().ConfigureAwait($true).GetAwaiter().GetResult()
+            $bytes = ([Microsoft.Identity.Client.ITokenCacheSerializer]$client.UserTokenCache).SerializeMsalV3()
         }
         else
         {
-            Write-Warning "Attempting to find a refresh token and an access token from the existing token cache data..."
+            $provider = [Microsoft.Azure.Commands.Common.Authentication.SharedTokenCacheProvider]::new()
+            $accounts = $provider.ListAccounts()
+            $bytes = $provider.ReadTokenData()
         }
-
-        #
-        # Resolve token cache data
-        #
-
-        [Microsoft.Azure.Commands.Common.Authentication.Authentication.Clients.AuthenticationClientFactory]$authenticationClientFactory = $null
-        if (-not ([Microsoft.Azure.Commands.Common.Authentication.AzureSession]::Instance.TryGetComponent(
-            [Microsoft.Azure.Commands.Common.Authentication.Authentication.Clients.AuthenticationClientFactory]::AuthenticationClientFactoryKey,
-            [ref]$authenticationClientFactory)))
+        if (-not $bytes)
         {
-            $m = 'Please ensure you have authenticated with Az Accounts module!'
-            $m += ' Unable to resolve authentication client factory from Az Accounts module runtime'
-            $m += ' ([Microsoft.Azure.Commands.Common.Authentication.Authentication.Clients.AuthenticationClientFactory])'
-            Write-Error $m
-            return
+            Write-Error "Ensure you enable context autosave for the process (using Enable-AzContextAutosave -Scope Process) and then login before calling this method."
```

Comment:
need to return after this

Reply:
Error: unable to resolve refresh token from empty context. Ensure you enable.....<>

Reply:
fixed